hack/run-hive-locally.sh

```diff
@@ -0,0 +1,296 @@
+#!/bin/bash
+
+# TODO: hiveadmission support
+# TODO: run hive-controllers & hive-clustersync separately with multiple replicas each (need docker to avoid metrics port collision)
+
+set -e
+set -u
+set -o pipefail
+
+# Highlighted echo
+function hl_echo {
+    tput setaf 3
+    echo "$@"
+    tput sgr0
+}
+
+function usage {
+    echo "The script runs a Hive component ($hive_component_operator or $hive_component_controllers) locally,
+simulating in-cluster environment. It records the current in-cluster Hive configurations,
+scales down the selected component, then runs it locally with the same environment.
+On pressing ctrl + c, the script is stopped and relevant components are scaled back up.
+
+Notes:
+- Artifacts (in-cluster Hive configurations and log files), if produced, are stored in the --artifacts_folder
+which is removed after the script ends if the --cleanup_artifacts is specified.
+- Selecting $hive_component_controllers runs all Hive controllers (including hive-clustersync) together
+as a single replica.
+
+Usage:
+- ./hack/run-hive-locally.sh [--flag=value ...]
+
+Supported flags:
+--hive-operator-ns=NAMESPACE
+--hive_operator_log_level=LOG_LEVEL
+--hive_controllers_log_level=LOG_LEVEL
+--artifacts_folder=DIRECTORY
+--cleanup_artifacts
+--help"
+}
+
+function kill_background_jobs {
+    while read -r pid; do
+        hl_echo "Killing job $pid"
+        kill "$pid" || true
+    done < <(jobs -p)
+    sleep "$kill_job_grace_period_seconds"
```

Comment:
Are you going to send a stronger signal once this grace period passes, or are you just giving time for processes to exit? If the latter, consider using `wait` instead.

Reply:
The latter. Fixed.